model/notification.go

```diff
@@ -1,12 +1,95 @@
 package model
 
+import (
+	"fmt"
+	"log"
+
+	"github.com/jmoiron/sqlx"
+)
+
 type Notification struct {
+	// From the notifications table.
 	UserID         int    `db:"user_id"`
 	AnnouncementID int    `db:"announcement_id"`
 	Status         string `db:"status"`
 	Receipt        string `db:"receipt"`
-	ReceiptStatus  string `db:"receipt_status"`
-	Timestamp      string `db:"timestamp"`
+	// From joined tables.
+	ExpoPushToken string `db:"expo_push_token"`
+	Title         string `db:"title"`
+	Body          string `db:"body"`
 }
 
-// TODO
+func EnqueueAnnouncementNotifications(db *sqlx.DB) {
+	log.Println("Starting to enqueue announcement notifications.")
+
+	tx := db.MustBegin()
+
+	insertQuery := `
+INSERT IGNORE into notifications (user_id, announcement_id, status)
+
+SELECT users.id as user_id, announcements.id as announcement_id, "queued" as status
+FROM announcements
+JOIN users ON users.conference_id = announcements.conference_id
+WHERE sent = 0 AND send_time <= NOW() AND expo_push_token is not null
+ORDER BY send_time asc
+`
+	results := tx.MustExec(insertQuery)
+	notificationRows, err := results.RowsAffected()
+	if err != nil {
+		fmt.Println("failed to insert announcement notifications")
+		tx.Rollback()
+		return
+	}
+
+	updateQuery := `
+UPDATE announcements
+SET sent = 1
+WHERE id in (SELECT DISTINCT announcement_id FROM notifications) AND sent = 0
+`
+	results = tx.MustExec(updateQuery)
+	announcementRows, err := results.RowsAffected()
+	if err != nil {
+		fmt.Println("failed to update announcements table")
+		tx.Rollback()
+		return
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		fmt.Println("failed to commit announcements to notifications table")
+		tx.Rollback()
```

Comment:
FYI, I don't think you need to call tx.Rollback() if tx.Commit() fails.

Reply:
Ok, thanks. Will remove that.